From 21a3f43c6888b216ab70ef9aff5c3903ed1f218f Mon Sep 17 00:00:00 2001
From: Ian Lance Taylor <iant@golang.org>
Date: Tue, 13 Sep 2022 10:33:20 -0700
Subject: [PATCH 064/547] runtime: ignore __morestack function in
 runtime.Callers

Backport from trunk.

We were ignoring all functions starting with "__morestack_", but not
the function "__morestack" itself.  Without this change, some tests
such as recover.go started failing recently, though I'm not sure
exactly what changed.

Reviewed-on: https://go-review.googlesource.com/c/gofrontend/+/427935
---
 libgo/runtime/go-callers.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libgo/runtime/go-callers.c b/libgo/runtime/go-callers.c
index 31783696a08..1d4dee66279 100644
--- a/libgo/runtime/go-callers.c
+++ b/libgo/runtime/go-callers.c
@@ -107,7 +107,7 @@ callback (void *data, uintptr_t pc, const char *filename, int lineno,
       p = function;
       if (__builtin_strncmp (p, "___", 3) == 0)
 	++p;
-      if (__builtin_strncmp (p, "__morestack_", 12) == 0)
+      if (__builtin_strncmp (p, "__morestack", 11) == 0)
 	return 0;
     }
   else if (filename != NULL)
-- 
2.25.1

